tests/testChain.cpp

```diff
@@ -0,0 +1,84 @@
+/* ----------------------------------------------------------------------------
+ * GTDynamics Copyright 2021, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file  testChain.cpp
+ * @brief Test Chain class.
+ * @author: Dan Barladeanu, Frank Dellaert
+ */
+
+#include <CppUnitLite/TestHarness.h>
+#include <gtsam/base/Matrix.h>
+#include <gtsam/geometry/Pose3.h>
+
+#include "gtdynamics/dynamics/Chain.h"
+
+using namespace gtdynamics;
+using gtsam::assert_equal;
+using gtsam::Matrix;
+using gtsam::Point3;
+using gtsam::Pose3;
+using gtsam::Rot3;
+
+// Test Chain class functionality with a three-joint chain
+TEST(threeLinks, All) {
+  // Initialize pose and screw axis for chain instantiation
+  Pose3 sMb = Pose3(Rot3(), Point3(5, 0, 0));
+  Matrix screwAxis(6, 1);
+  screwAxis << 0.0, 0.0, 1.0, 0.0, 5.0, 0.0;
+
+  // Instantiate chains and create a vector
+  Chain joint1(sMb, screwAxis), joint2(sMb, screwAxis), joint3(sMb, screwAxis);
+  std::vector<Chain> chains{joint1, joint2, joint3};
+
+  // Compose chains
+  Chain composed = Chain::compose(chains);
+
+  // Check result of composition
+  Pose3 expected = Pose3(Rot3(), Point3(15, 0, 0));
+  Matrix expected_J(6, 3);
+  expected_J << 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 15, 10, 5, 0, 0, 0;
+  EXPECT(assert_equal(composed.sMb(), expected, 1e-6));
+  EXPECT(assert_equal(composed.axes(), expected_J, 1e-6));
+
+  // Test POE functionality on FK at rest (without jacobian)
+  Vector joint_angles(3);
+  joint_angles << 0, 0, 0;
+  Pose3 POE = composed.poe(joint_angles);
+  EXPECT(assert_equal(POE, expected, 1e-6));
+
+  // Test POE functionality on FK at rest (with jacobian)
+  Matrix J0;
+  POE = composed.poe(joint_angles, boost::none, J0);
+  EXPECT(assert_equal(POE, expected, 1e-6));
+  EXPECT(assert_equal(J0, expected_J, 1e-6));
+
+  // Test POE functionality on FK not at rest
```

Comment:
This whole test should be broken up into 4 tests for each of these blocks.

You could just define each chain as a global variable (if they are not modified), or copy paste them into each test.

Reply:
done